src/ReadProperties.php

```diff
@@ -0,0 +1,158 @@
+ * using \Robo\Common\ConfigAwareTrait.
+ *
+ * ``` php
+ * $this->taskReadProperties()
+ *     ->in(getcwd())
+ *     ->followSymlinks()
+ *     ->run();
+ * $this->getConfig()->get('my.config.setting');
+ * ```
+ *
+ */
+class ReadProperties extends BaseTask
+{
+
+    /**
+     * The finder object to scan for files.
+     *
+     * @var \Symfony\Component\Finder\Finder
+     */
+    protected $finder;
+
+    /**
+     * The directories to search in.
+     *
+     * @var array
+     */
+    protected $dirs;
+
+    /**
+     * Searches files and directories which match defined rules.
+     *
+     * @param string|array $dirs
+     *   A directory path or an array of directories in which to search for
+     *   config files.
+     *
+     * @throws \InvalidArgumentException
+     *   If one of the directories does not exist.
+     */
+    public function __construct($dirs = [])
+    {
+        $this->finder = new Finder();
+        $this->dirs = $dirs;
+    }
+
+    /**
+     * Sets the finder.
+     *
+     * @param \Symfony\Component\Finder\Finder $finder
+     *
+     * @return $this
+     *
+     * @codeCoverageIgnore
+     */
+    public function finder(Finder $finder)
+    {
+        $this->finder = $finder;
+
+        return $this;
+    }
+
+    /**
+     * Searches files and directories which match defined rules.
+     *
+     * @param string|array $dirs
+     *   A directory path or an array of directories in which to search for
+     *   config files.
+     *
+     * @return $this
+     *
+     * @throws \InvalidArgumentException
+     *   If one of the directories does not exist.
+     *
+     * @codeCoverageIgnore
+     */
+    public function in($dirs)
+    {
+        $this->dirs = $dirs;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function run()
+    {
+        try {
+            $this->finder->in($this->dirs)->files();
+            $defaults = clone $this->finder;
+            $packageOverrides = clone $this->finder;
+            $projectConfig = [];
+
+            // Get the property overrides for this project.
+            $root = $this->getConfig()->get('digipolis.root.project', false);
+            if ($root && file_exists($root . '/properties.yml')) {
+                $this->logger()->debug('Parsing config from ' . $root . '/properties.yml.');
+                $projectConfig = Yaml::parse(file_get_contents($root . '/properties.yml'));
+            }
+
+            $parsedConfig = array_merge(
+                // Get the default properties.
+                $this->parseConfigFiles($defaults->name('default.properties.yml')),
+                // Get the property overrides for robo packages.
+                $this->parseConfigFiles($packageOverrides->name('properties.yml')),
+                // Add the project overrides last.
+                $projectConfig
+            );
+            // @todo: Add properties from command line arguments?
```

Comment:
I was thinking about this, and I think this @todo can go, without implementing anything new.

This is a task, task are used by commands. Tasks do not receive command line arguments... commands do. So it's up to the implementation of the command to figure out if it wants to receive properties from the command line arguments and overwrite the properties that were read from the files IMHO.